lib/tests/test_sweeps.c

```diff
@@ -307,6 +307,77 @@ test_sweep_genic_selection_time_change(void)
     tsk_table_collection_free(&tables);
 }
 
+static void
+sweep_genic_selection_reproduce_bug_1111_single_run(void)
+{
+
+    /* Try to mimic the msms parameters used in verification.py
+           "100 300 -t 200 -r 200 500000"
+           " -SF 0 0.9 -Sp 0.5 -SaA 5000 -SAA 10000 -N 10000"
+       */
+    int ret;
+    uint32_t n = 100;
+    double num_loci = 500001;
+    double position = 0.5;
+    double alpha = 10000;
+    double recom_rate = 0.0004;
+    double start_frequency = 0.5 / 10000;
+    double end_frequency = 0.9;
+    double dt = 1.0 / 400000;
+
+    msp_t msp;
+    sample_t *samples = malloc(n * sizeof(sample_t));
+    gsl_rng *rng = gsl_rng_alloc(gsl_rng_default);
+    tsk_table_collection_t tables;
+
+    gsl_rng_set(rng, 1000);
```

Comment:
Ah, so you're using the same random seed here each time. Try making ``seed`` a parameter and passing in your index variable i in the function below.

Reply:
I tried this method which always leads to crashes.
```
Suite: msprime
  Test: test_genic_selection_trajectory ...passed
  Test: test_sweep_genic_selection_bad_parameters ...passed
  Test: test_sweep_genic_selection_events ...passed
  Test: test_sweep_genic_selection_single_locus ...passed
  Test: test_sweep_genic_selection_recomb ...passed
  Test: test_sweep_genic_selection_time_change ...passed
  Test: test_sweep_genic_selection_reproduce_bug_1111 ...seed: 1seed: 2seed: 3seed: 4seed: 5seed: 6seed: 7seed: 8seed: 9seed: 10seed: 11seed: 12seed: 13seed: 14seed: 15seed: 16seed: 17seed: 18seed: 19seed: 20seed: 21seed: 22seed: 23seed: 24seed: 25seed: 26seed: 27seed: 28seed: 29seed: 30seed: 31seed: 32seed: 33seed: 34seed: 35bug!gsl: ../gsl/gsl_rng.h:199: ERROR: invalid n, either 0 or exceeds maximum value of generator
Default GSL error handler invoked.

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff79f2537 in __GI_abort () at abort.c:79
#2  0x00007ffff7c2311d in gsl_error () from /lib/x86_64-linux-gnu/libgsl.so.25
#3  0x00007ffff7cf340d in gsl_rng_uniform_int () from /lib/x86_64-linux-gnu/libgsl.so.25
#4  0x0000555555568a56 in msp_std_common_ancestor_event (self=0x7fffffffe1d0, population_id=0, label=1) at ../msprime.c:6006
#5  0x0000555555566179 in msp_run_sweep (self=0x7fffffffe1d0) at ../msprime.c:4782
#6  0x00005555555664e8 in msp_run (self=0x7fffffffe1d0, max_time=1.7976931348623157e+308, max_events=4294967295) at ../msprime.c:4879
#7  0x000055555555891f in sweep_genic_selection_reproduce_bug_1111_single_run (seed=35) at ../tests/test_sweeps.c:363
#8  0x00005555555589c1 in test_sweep_genic_selection_reproduce_bug_1111 () at ../tests/test_sweeps.c:379
#9  0x00007ffff7faba97 in ?? () from /lib/x86_64-linux-gnu/libcunit.so.1
#10 0x00007ffff7fabd39 in ?? () from /lib/x86_64-linux-gnu/libcunit.so.1
#11 0x00007ffff7fac137 in CU_run_all_tests () from /lib/x86_64-linux-gnu/libcunit.so.1
#12 0x0000555555586e0e in test_main (tests=0x7fffffffe7a0, argc=1, argv=0x7fffffffe918) at ../tests/testlib.c:97
#13 0x0000555555558aaf in main (argc=1, argv=0x7fffffffe918) at ../tests/test_sweeps.c:400
```

There seems to be a bug in `msp_std_common_ancestor_event` function. Some seeds can lead to the  `n-1` equals  `0` and then gsl error.

`j = (uint32_t) gsl_rng_uniform_int(self->rng, (n - 1)); `